src/pages/NewPage.tsx

```diff
@@ -0,0 +1,164 @@
+  Heading,
+  HStack,
+  Icon,
+  Stack,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import { ReactNode, useCallback, useRef } from "react";
+import { RiAddLine, RiFolderOpenLine, RiRestartLine } from "react-icons/ri";
+import { FormattedMessage, useIntl } from "react-intl";
+import { useNavigate } from "react-router";
+import DefaultPageLayout, {
+  HomeMenuItem,
+  HomeToolbarItem,
+} from "../components/DefaultPageLayout";
+import LoadProjectInput, {
+  LoadProjectInputRef,
+} from "../components/LoadProjectInput";
+import NewPageChoice from "../components/NewPageChoice";
+import { useConnectionStage } from "../connection-stage-hooks";
+import { SessionPageId } from "../pages-config";
+import { useStore } from "../store";
+import { createSessionPageUrl } from "../urls";
+
+const NewPage = () => {
+  const existingSessionTimestamp = useStore((s) => s.timestamp);
+  const projectName = useStore((s) => s.project.header?.name ?? "Untitled");
+  const newSession = useStore((s) => s.newSession);
+  const navigate = useNavigate();
+  const { actions: connStageActions } = useConnectionStage();
+
+  const handleOpenLastSession = useCallback(() => {
+    navigate(createSessionPageUrl(SessionPageId.DataSamples));
+  }, [navigate]);
+
+  const loadProjectRef = useRef<LoadProjectInputRef>(null);
+  const handleContinueSessionFromFile = useCallback(() => {
+    loadProjectRef.current?.chooseFile();
+  }, []);
+
+  const handleStartNewSession = useCallback(() => {
+    newSession();
+    navigate(createSessionPageUrl(SessionPageId.DataSamples));
+    connStageActions.startConnect();
+  }, [newSession, navigate, connStageActions]);
+
+  const intl = useIntl();
+  const lastSessionTitle = intl.formatMessage({
+    id: "newpage-last-session-title",
+  });
+  const continueSessionTitle = intl.formatMessage({
+    id: "newpage-continue-session-title",
+  });
+  const newSessionTitle = intl.formatMessage({
+    id: "newpage-new-session-title",
+  });
+
+  return (
+    <DefaultPageLayout
+      toolbarItemsRight={<HomeToolbarItem />}
+      menuItems={<HomeMenuItem />}
+    >
+      <LoadProjectInput ref={loadProjectRef} accept=".json,.hex" />
+      <VStack alignItems="center">
+        <Container maxW="1180px" alignItems="stretch" zIndex={1} p={4} mt={8}>
+          <VStack alignItems="stretch" w="100%">
+            <Heading as="h1" fontSize="4xl" fontWeight="bold">
+              <FormattedMessage id="newpage-title" />
+            </Heading>
+            <Heading as="h2" fontSize="2xl" mt={8}>
+              <FormattedMessage id="newpage-section-one-title" />
+            </Heading>
+            <HStack
+              w="100%"
+              gap={8}
+              alignItems="stretch"
+              mt={3}
+              flexDir={{ base: "column", lg: "row" }}
+            >
+              <NewPageChoice
+                onClick={handleOpenLastSession}
+                label={lastSessionTitle}
+                disabled={!existingSessionTimestamp}
```

Comment:
When there's an existing session, the "Open last session" button appears disabled and navigates to the data samples page.
When there's not an existing session, the "Open last session" button appears enabled and navigates to the data samples page.

Maybe the expected behaviour should be: 
When there's an existing session, the "Open last session" button appears enabled and navigates to the data samples page.
When there's not an existing session, the "Open last session" button appears disabled and cannot be clicked?

Reply:
6bc333890966cda60cb1bdc3dcdae8dc6469a470
ca5b34c9d1175e6d19514c2ef6e15d6262757e49